2022/vi/Days/day74.md

```diff
@@ -0,0 +1,93 @@
+---
+title: '#90DaysOfDevOps - Hello World - Jenkinsfile App Pipeline - Day 74'
+published: false
+description: 90DaysOfDevOps - Hello World - Jenkinsfile App Pipeline
+tags: 'devops, 90daysofdevops, learning'
+cover_image: null
+canonical_url: null
+id: 1048744
+---
+
+## Hello World - Jenkinsfile App Pipeline
+
+Ở bài trước, chúng ta đã xây một Pipeline đơn giản trong Jenkins có thể đẩy ảnh docker từ dockerfile trong một kho lưu trữ Github công khai tới kho lưu trữ Dockerhub riêng tư.
+
+Trong bài này, chúng ta muốn bước một bước xa hơn và đạt được các mục tiêu sau với ứng dụng đơn giản của chúng ta.
+### Mục tiêu
+
+- Dockerfile (Hello World)
+- Jenkinsfile
+- Jenkins Pipeline để kích hoạt khi Kho lưu trữ GitHub được cập nhật
+- Sử dụng Kho lưu trữ GitHub làm nguồn.
+- Chạy - Sao chép/Lấy Kho lưu trữ, Xây dựng, Kiểm tra, Triển khai Các giai đoạn
+- Triển khai lên DockerHub với số phiên bản tăng dần
+- Mục tiêu mở rộng để triển khai lên Cụm Kubernetes của chúng ta (Điều này sẽ liên quan đến công việc và kho manifest khác sử dụng thông tin xác thực GitHub)
+
+### Bước một
+
+Chúng ta có [GitHub repository](https://github.com/MichaelCade/Jenkins-HelloWorld) đang chứa Dockerfile và tệp index.html
+![](../../Days/Images/Day74_CICD1.png)
+
+Với nội dung trên, đây là những gì chúng ta đang sử dụng làm nguồn trong Pipeline của mình, giờ đây chúng ta cũng muốn thêm kịch bản Jenkins Pipeline đó vào kho GitHub của chúng ta.
+
+![](../../Days/Images/Day74_CICD2.png)
+
+Hãy quay trở lại bảng điều khiển Jenkins, chúng ta sẽ tạo một pipeline mới nhưng thay vì sử kịch bản của chúng ta, chúng ta sẽ sử dụng "Pipeline script from SCM" và các tùy chọn cấu hình dưới đây.
+
+Để tham khảo, chúng ta sẽ sử dụng `https://github.com/MichaelCade/Jenkins-HelloWorld.git` làm URL của kho lưu trữ.
+
+![](../../Days/Images/Day74_CICD3.png)
+
+Ở bước này, chúng ta có thể nhấn "Lưu" và "Áp dụng", sau đó chúng ta sẽ có thể chạy thủ công Pipeline của mình để xây dựng hình ảnh Docker mới và tải lên kho DockerHub của chúng ta.
+Tuy nhiên, tôi cũng muốn đảm bảo rằng chúng ta đã đặt lịch mỗi khi kho lưu trữ hoặc mã nguồn thay đổi sẽ kích hoạt một tiến trình xây dựng. Chúng ta có thể sử dụng webhooks hoặc đồng bộ kéo định kỳ.
+
+Điều này đáng để quan tâm vì nếu bạn đang sử dụng các tài nguyên đám mây mất phí để giữ pipeline và bạn có nhiều thay đổi tới kho lưu trữ mã nguồn bạn sẽ chịu phát sinh nhiều chi phí. Chúng ta biết rằng đây chỉ là một môi trường chạy thử đó là lí do tôi đang sử dụng lựa chọn "poll scm". (Tôi cũng tin rằng sử dụng minikube tôi sẽ thiếu đi khả năng sử dụng webhooks )
```

Comment:
để giữ pipeline? not a very good expression, could you change it to something else?

Reply:
tôi sẽ thiếu đi khả năng sử dụng webhooks
-> tôi sẽ không có chức năng webhooks 

is this better?